.github/workflows/test.yml

```diff
@@ -30,6 +30,7 @@ jobs:
           - "test.run.test_sources"
           - "test.run.test_stages"
           - "stages/test"
```

Comment:
Can/could be a followup of course!

Reply:
I wonder if something simple as
```yaml
diff --git a/.github/workflows/test.yml b/.github/workflows/test.yml
index 1e6be937..f75452b2 100644
--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -21,15 +21,8 @@ jobs:
       fail-fast: false
       matrix:
         test:
-          - "test.mod"
-          - "test.run.test_boot"
-          - "test.run.test_devices"
-          - "test.run.test_executable"
-          - "test.run.test_mount"
-          - "test.run.test_noop"
-          - "test.run.test_sources"
           - "test.run.test_stages"
-          - "stages/test"
+          - "-k not test.run.test_stages"
         environment:
           - "py36"    # RH8
           - "py39"    # RH9
````
might work

Reply:
I think that'd be nice.  It would avoid the issue we have now of not picking up new tests.  If a new test should be excluded, it's better to have it fail and realise that we should add it.

Reply:
> I wonder if something simple as
> 
> ```yaml
> diff --git a/.github/workflows/test.yml b/.github/workflows/test.yml
> index 1e6be937..f75452b2 100644
> --- a/.github/workflows/test.yml
> +++ b/.github/workflows/test.yml
> @@ -21,15 +21,8 @@ jobs:
>        fail-fast: false
>        matrix:
>          test:
> -          - "test.mod"
> -          - "test.run.test_boot"
> -          - "test.run.test_devices"
> -          - "test.run.test_executable"
> -          - "test.run.test_mount"
> -          - "test.run.test_noop"
> -          - "test.run.test_sources"
>            - "test.run.test_stages"
> -          - "stages/test"
> +          - "-k not test.run.test_stages"
>          environment:
>            - "py36"    # RH8
>            - "py39"    # RH9
> ```
> 
> might work

Hang on, I'm confused.  The current matrix is
```yaml
      matrix:
        test:
          - "test.mod"
          - "test.run.test_boot"
          - "test.run.test_devices"
          - "test.run.test_executable"
          - "test.run.test_mount"
          - "test.run.test_noop"
          - "test.run.test_sources"
          - "test.run.test_stages"
          - "stages/test"
        environment:
          - "py36"    # RH8
          - "py39"    # RH9
          - "py312"   # latest fedora
        # special keyword "include"
        include:
          # this test _can_ run in parallel
          # Using 4 workers is a bit arbitrary, "auto" is probably too aggressive.
          - test: "test.run.test_stages"
            env_TEST_WORKERS: "-n 4"
```
so `test.run.test_stages` appears both in the `matrix.test` and the `include` section.  Is that right?  Or were we running stages twice?

Reply:
@mvo5 ok now I see how this works.  One issue with changing it like this is that each line is a separate job, so this becomes a single run, whereas before we were running them all separately and in parallel.

Reply:
Perhaps we can have a check somewhere that all tests are in the matrix, or have it generated by a script like we do with the gitlab-ci.yml in images.

Reply:
Thanks! Yeah, let's land it, sorry for complicating things :)